### PR TITLE
End your search

### PR DESCRIPTION
**Pull request by DilwoarH**
As a G-Cloud buyer
I need a fixed list of services
so that I can start the shortlisting process

Ticket: https://trello.com/c/gs2I3Cj2/676-end-your-search

### CODE REVIEW

File: app/assets/scss/_end-your-search-page.scss [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.end-search-page {
```
Comment (TheDoubleK) [inline app/assets/scss/_end-your-search-page.scss]: Could this go in the buttons section of the front-end toolkit?

Comment (TheDoubleK) [reply app/assets/scss/_end-your-search-page.scss]: Perhaps with a more generic name, to allow for reuse.

File: app/templates/direct-award/end-search.html [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      {
+        "link": "/buyers",
+        "label": "Your Account"
+      },
+      {
+        "link": "/buyers/direct-award",
+        "label": "Saved Searches"
+      },
+      {
+        "link": "/buyers/direct-award/{}".format(project.id),
```
Comment (samuelhwilliams) [inline app/templates/direct-award/end-search.html]: These should be `url_for`d

File: app/assets/scss/_end-your-search-page.scss (hunk 1 above)
Comment (galund) [reply app/assets/scss/_end-your-search-page.scss]: We're gonna go with the standard (bigger) button, anyway :) - thanks

File: app/templates/direct-award/end-search.html [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  <p>download the shortlist to help you compare services.</p>
+
+  
+  <br>
+  <form 
+    action="{{ url_for('direct_award.end_search_action', framework_framework=framework.framework, project_id=project.id) }}"
+    method="POST"  
+    >
+    <input type="hidden" name="csrf_token" value="{{ csrf_token() }}" /> 
+    <button class="button-save end-your-searxch-button">End search and continue</button>
```
Comment (samuelhwilliams) [inline app/templates/direct-award/end-search.html]: `searxch`

File: app/main/views/g_cloud.py [hunk 4]
```diff
@@ -34,6 +34,7 @@
 
 
 PROJECT_CREATED_MESSAGE = Markup("""Your new project has been created.""")
+PROJECT_ENDED_MESSAGE = Markup("""Your search has been ended. You can now download your shortlist.""")
```
Comment (samuelhwilliams) [inline app/main/views/g_cloud.py]: Is this real content? Seems like a weird sentence.

File: app/main/views/g_cloud.py [hunk 5]
```diff
@@ -292,7 +293,7 @@ def saved_search_overview(framework_framework):
     closed_projects = []
 
     for project in projects:
-        if project['active']:
+        if project['lockedAt'] is None:
```
Comment (galund) [inline app/main/views/g_cloud.py]: Let's put this back - the distinction for this project list page is between open procurements and closed ones. Saved searches are still 'open' in this sense - they have neither been awarded nor archived, so they are 'active'.

if we want to show the locked status here, then that's a different thing?

File: app/templates/direct-award/end-search.html [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  </header>
+
+  <div class="dmspeak">
+    <h2 class="heading-xmedium">Before you end your search</h2>
+  </div>
+  <p>You should only continue when you have finished searching for cloud services</p>
+  <p>and are ready to start comparing the services you've found</p>
+
+  <br>
+  <p>You can not edit your search after it has ended.</p>
```
Comment (galund) [inline app/templates/direct-award/end-search.html]: "Can't" (or "cannot"), rather than "can not" I think?

File: app/main/views/g_cloud.py (hunk 4 above)
Comment (galund) [reply app/main/views/g_cloud.py]: Placeholder, real thing can edit for Karl, agree first sentence should probably have been "marked as complete" or something, but somehow be consistent with "ending".. shrug :)

File: app/templates/direct-award/end-search.html [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+      End search
+    </h1>
+  </header>
+
+  <div class="dmspeak">
+    <h2 class="heading-xmedium">Before you end your search</h2>
+  </div>
+  <p>You should only continue when you have finished searching for cloud services</p>
+  <p>and are ready to start comparing the services you’ve found</p>
+
```
Comment (galund) [inline app/templates/direct-award/end-search.html]: There's an errant paragraph el here

File: app/templates/errors/400.html [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  <div class="index-page grid-row">
+    <div class="column-two-thirds">
+      <header class="page-heading-smaller">
+        <h1>Sorry, there was a problem with your request</h1>
+      </header>
+      <p>
+        {% if error_message %}
+          {{ error_message }}
+        {% else %}
+          Please do not attempt the same request again.
```
Comment (galund) [inline app/templates/errors/400.html]: Assume there's only contrived ways for this to happen, but what comes to mind is multiple user concurrency, so when we add collaboration we could think about how to make the recovery more friendly (e.g. provide a link back to the project)

Comment (samuelhwilliams) [reply app/templates/errors/400.html]: This was mostly copied from the user-frontend, but yeah we should sort out the content eventually.

We should bring all the error message templates into the frontend-toolkit eventually. I started doing that but it seems like a non-trivial thing to link them all up from a central place, so shelved it for now. I dropped a ticket in the tech debt board a month or two ago about it anyway.

File: app/templates/direct-award/end-search.html (hunk 7 above)
Comment (samuelhwilliams) [reply app/templates/direct-award/end-search.html]: Good spot; we were missing a couple of div wrappers to get the proper two-thirds styling. Have added.

File: app/templates/errors/400.html (hunk 8 above)
Comment (samuelhwilliams) [reply app/templates/errors/400.html]: Regarding contrived ways for this to happen: if you go back in the browser you can hit this page again, and in Chrome it automatically re-sends the request so it's not that difficult to conceivably hit this 400 as a fairly average user. But yeah we could probably have something nicer once collaboration comes in.

**Pull request merged**